app/gui/src/dashboard/pages/dashboard/UserBar/UserBar.tsx

```diff
@@ -70,6 +82,37 @@ export function UserBar(props: UserBarProps) {
   const { getText } = useText()
   const { isFeatureUnderPaywall } = usePaywall({ plan: user.plan })
   const { isOffline } = useOffline()
+  const { remoteBackend } = useBackends()
+  const { data: organization } = useQuery(
+    backendQueryOptions(remoteBackend, 'getOrganization', [], {
+      enabled: user.isOrganizationAdmin,
+    }),
+  )
+  const trialEndDate =
+    user.isOrganizationAdmin && organization?.subscription?.trialEnd != null ?
+      new Date(organization.subscription.trialEnd)
+    : null
+  const trialStartDate =
+    user.isOrganizationAdmin && organization?.subscription?.trialStart != null ?
+      new Date(organization.subscription.trialStart)
+    : null
+  const msToTrialEnd = Number(trialEndDate) - Number(new Date())
+  const trialDaysLeft = trialEndDate ? Math.max(0, Math.floor(msToTrialEnd / DAY_MS)) : null
+  const trialHoursLeft =
+    trialDaysLeft != null && trialDaysLeft < 1 ?
+      Math.max(0, Math.floor(msToTrialEnd / HOUR_MS))
+    : null
+  const trialProgress =
+    trialStartDate != null && trialEndDate != null ?
+      1 - msToTrialEnd / (Number(trialEndDate) - Number(trialStartDate))
+    : null
+  const trialText =
+    trialDaysLeft == null ? null
+    : trialHoursLeft != null ?
+      trialHoursLeft > 0 ?
+        getText('xDaysLeftInTrial', trialHoursLeft)
+      : getText('lessThanOneHourLeftInTrial')
+    : getText('xDaysLeftInTrial', trialDaysLeft)
```

Comment:
This logic deserves it's own function. It could return object with "days left", "hours left" etc., and early returns will make it looking nicer. It shoulc be also unit tested.